tests/components/flux_led/test_config_flow.py

```diff
@@ -0,0 +1,450 @@
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch(
+        "homeassistant.components.flux_led.async_setup",
+        return_value=True,
+    ) as mock_setup, patch(
+        "homeassistant.components.flux_led.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_AUTOMATIC_ADD: False,
+                CONF_DEVICES: {
+                    "1_1_1_1": {
+                        CONF_NAME: "TestLight",
+                        CONF_HOST: "1.1.1.1",
+                    }
+                },
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "FluxLED/MagicHome"
+    assert result2["data"] == {
+        CONF_AUTOMATIC_ADD: False,
+        CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+        CONF_DEVICES: {
+            "1_1_1_1": {
+                CONF_NAME: "TestLight",
+                CONF_HOST: "1.1.1.1",
+            }
+        },
+    }
+
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_manual_already_setup(hass):
+    """Test import manual added when the device is already configured."""
+    MockConfigEntry(
+        domain="flux_led",
+        data={
+            CONF_AUTOMATIC_ADD: False,
+            CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+            CONF_DEVICES: {
+                "1_1_1_1": {
+                    CONF_NAME: "TestLight",
+                    CONF_HOST: "1.1.1.1",
+                }
+            },
+        },
+    ).add_to_hass(hass)
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_IMPORT},
+        data={
+            CONF_AUTOMATIC_ADD: False,
+            CONF_DEVICES: {
+                "1_1_1_1": {
+                    CONF_NAME: "TestLight",
+                    CONF_HOST: "1.1.1.1",
+                }
+            },
+        },
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_options_set_global_options(hass):
+    """Test set global options through options flow."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    entry = MockConfigEntry(
+        domain="flux_led",
+        data={
+            CONF_AUTOMATIC_ADD: False,
+            CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+            CONF_DEVICES: {
+                "1_1_1_1": {
+                    CONF_NAME: "TestLight",
+                    CONF_HOST: "1.1.1.1",
+                }
+            },
+        },
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
We need to set up the integration before starting the options flow. Use `hass.config_entries.async_setup`.

Reply:
Isn't that what this does:

```
    await setup.async_setup_component(hass, "persistent_notification", {})

    entry = MockConfigEntry(
        domain="flux_led",
        data={
            CONF_AUTOMATIC_ADD: False,
            CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
            CONF_DEVICES: {
                "1_1_1_1": {
                    CONF_NAME: "TestLight",
                    CONF_HOST: "1.1.1.1",
                }
            },
        },
    )
    entry.add_to_hass(hass)
```

Reply:
No, that just adds the config entry to the list of entries.

Reply:
OK - I based my test flow of the rfxtrx test flow. Can you point me to a component which does this correctly so I can better understand what's missing?

Reply:
https://github.com/home-assistant/core/blob/6d043f2ca12ca6e218c86e168f40da4fe5348fd2/tests/components/accuweather/test_config_flow.py#L145-L168

Reply:
OK I think I've got it.